rockcraft/models/project.py

```diff
@@ -281,6 +254,7 @@ class Project(YamlModelMixin, BuildPlanner, BaseProject):  # type: ignore[misc]
     services: dict[str, Service] | None
     checks: dict[str, Check] | None
     entrypoint_service: str | None
+    platforms: dict[str, Platform | None]  # type: ignore[assignment]
```

Comment:
The generated schema is the same, with or without ` | None`.  Either way, this job fails:
https://github.com/canonical/rockcraft/actions/runs/9289669392/job/25564121553?pr=567#step:5:11

Reply:
fixed the schema

Reply:
Ah, I wasn't aware you could manually tweak the schema